migration/migrator/migrations/course/20190528164528_numeric_id.py

```diff
@@ -0,0 +1,34 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE IF EXISTS users ADD COLUMN user_numeric_id varchar;")
```

Comment:
As [the document on migrations](https://submitty.org/developer/migrations) suggests:

> it is important to ensure that the up migration can be re-run. For example, your up function should not crash on adding the column if the column already exists.

It can be good to change `ADD COLUMN` to `ADD COLUMN IF NOT EXISTS` and `CREATE INDEX` to `CREATE INDEX IF NOT EXISTS`, or adding an `if` to determine if there is already a `user_numeric_id` column.

Reply:
It's good to bring this up.  However, this also brings up other issues in my mind...
* Presumably, this migration would only crash when the column already exists, but the migration ran anyway.  That implies there is no migration record to disallow another migration, so something else outside of this migration went awry.
* The migration script that is calling these functions probably should be doing exception handling on migration functions.  That would be far more efficient than having every migration function do its own exception handling.  That is not to say that the migrator should ignore bad DB queries.  But the migrator can note a bad query and cleanly post an error message to console and/or log.  And then decide to proceed to the next migration, or not.

But these cases are out of scope for this PR.  I just felt they were worth mentioning.

Reply:
Change made.  Parallel change also made to master DB migration.